torch/autograd/function.py

```diff
@@ -8,24 +8,54 @@
 from collections import OrderedDict
 from typing import Any, List, Optional
 
+# Formerly known as: _ContextMethodMixin
+# Any class subclassing Function will
+class Ctx(object):
 
-class _ContextMethodMixin(object):
-
-    def save_for_backward(self, *tensors):
+    def save_for_backward(self, *tensors: torch.Tensor):
         r"""Saves given tensors for a future call to :func:`~Function.backward`.
 
         **This should be called at most once, and only from inside the**
-        :func:`forward` **method.**
+        :func:`forward` **method. This should only be called with input or
+        output tensors**
 
-        Later, saved tensors can be accessed through the :attr:`saved_tensors`
+        In :func:`backward`, saved tensors can be accessed through the :attr:`saved_tensors`
         attribute. Before returning them to the user, a check is made to ensure
         they weren't used in any in-place operation that modified their content.
 
         Arguments can also be ``None``.
```

Comment:
nit maybe mention here that this is a no-op in this case?

Reply:
Why do we allow `None` arguments anyway? Only for BC reasons I'm guessing? Doesn't seem very useful...

Reply:
No good reason I know of.